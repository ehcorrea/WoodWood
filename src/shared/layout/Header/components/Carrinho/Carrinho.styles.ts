import styled, { css } from '@emotion/native';
import Animated from 'react-native-reanimated';

export const Wrapper = styled(Animated.View)`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.white.main};
    border-radius: 10px;
    height: 35px;
    justify-content: center;
    width: 35px;
  `}
`;

export const ContainerValor = styled.View`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.error.main};
    border-radius: 20px;
    height: 20px;
    justify-content: center;
    min-width: 30px;
    padding-top: 2px;
    position: absolute;
    right: -10px;
    top: -12px;
  `}
`;
