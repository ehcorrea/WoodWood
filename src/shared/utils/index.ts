export * from './carrinho/carrinho';
export * from './responsive/responsive';
