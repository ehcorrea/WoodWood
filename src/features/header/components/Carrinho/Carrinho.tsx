import { Keyframe, LinearTransition } from 'react-native-reanimated';

import { IconCart } from '@/assets/icons';
import { Text } from '@/shared/components';

import * as S from './Carrinho.styles';

type CarrinhoProps = {
  valor: number;
};

const Tilt = new Keyframe({
  0: {
    transform: [{ rotate: '-10deg' }],
    opacity: 0.7,
  },
  50: {
    transform: [{ rotate: '10deg' }],
  },
  100: {
    transform: [{ rotate: '0deg' }],
    opacity: 1,
  },
}).duration(400);

export function Carrinho({ valor }: CarrinhoProps) {
  return (
    <S.Wrapper
      key={valor}
      entering={valor ? Tilt : undefined}
      layout={LinearTransition.springify()}
    >
      <IconCart width={30} height={30} />
      <S.ContainerValor>
        <Text palette="white" weight="semibold" size="small">
          {valor >= 100 ? '+99' : valor}
        </Text>
      </S.ContainerValor>
    </S.Wrapper>
  );
}
