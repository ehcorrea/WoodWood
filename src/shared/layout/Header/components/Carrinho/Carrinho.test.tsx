import { render } from '@/test/utils';
import { Carrinho } from './Carrinho';

const setup = (valor: number = 10) => {
  return render(<Carrinho valor={valor} />);
};

describe('<Carrinho/>', () => {
  describe('quando renderizado', () => {
    test('e possui valor menor que 99', () => {
      const container = setup();
      expect(container.getByText('10')).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    test('e possui valor maior que 99', () => {
      const container = setup(100);
      expect(container.getByText('+99')).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
