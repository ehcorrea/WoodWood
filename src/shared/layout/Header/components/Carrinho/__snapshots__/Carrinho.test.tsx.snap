// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carrinho/> quando renderizado e possui valor maior que 99 1`] = `
<View
  collapsable={false}
  entering={
    InnerKeyframe {
      "build": [Function],
      "definitions": {
        "0": {
          "opacity": 0.7,
          "transform": [
            {
              "rotate": "-10deg",
            },
          ],
        },
        "100": {
          "opacity": 1,
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        },
        "50": {
          "transform": [
            {
              "rotate": "10deg",
            },
          ],
        },
      },
      "durationV": 400,
      "parsedAnimation": [Function],
      "reduceMotionV": "system",
    }
  }
  jestAnimatedProps={
    {
      "value": {},
    }
  }
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  jestInlineStyle={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": 35,
        "justifyContent": "center",
        "width": 35,
      },
      undefined,
    ]
  }
  layout={
    LinearTransition {
      "build": [Function],
      "durationV": undefined,
      "randomizeDelay": false,
      "reduceMotionV": "system",
      "type": [Function],
    }
  }
  nativeID="1"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": 35,
        "justifyContent": "center",
        "width": 35,
      },
      undefined,
    ]
  }
>
  <svg
    data-file-name="SvgIconCart"
    height={30}
    width={30}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F30000",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "height": 20,
          "justifyContent": "center",
          "minWidth": 30,
          "paddingTop": 2,
          "position": "absolute",
          "right": -10,
          "top": -12,
        },
        undefined,
      ]
    }
  >
    <Text
      palette="white"
      size="small"
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "SofiaPro-SemiBold",
            "fontSize": 17,
            "lineHeight": 17,
          },
          undefined,
        ]
      }
      weight="semibold"
    >
      +99
    </Text>
  </View>
</View>
`;

exports[`<Carrinho/> quando renderizado e possui valor menor que 99 1`] = `
<View
  collapsable={false}
  entering={
    InnerKeyframe {
      "build": [Function],
      "definitions": {
        "0": {
          "opacity": 0.7,
          "transform": [
            {
              "rotate": "-10deg",
            },
          ],
        },
        "100": {
          "opacity": 1,
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        },
        "50": {
          "transform": [
            {
              "rotate": "10deg",
            },
          ],
        },
      },
      "durationV": 400,
      "parsedAnimation": [Function],
      "reduceMotionV": "system",
    }
  }
  jestAnimatedProps={
    {
      "value": {},
    }
  }
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  jestInlineStyle={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": 35,
        "justifyContent": "center",
        "width": 35,
      },
      undefined,
    ]
  }
  layout={
    LinearTransition {
      "build": [Function],
      "durationV": undefined,
      "randomizeDelay": false,
      "reduceMotionV": "system",
      "type": [Function],
    }
  }
  nativeID="0"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": 35,
        "justifyContent": "center",
        "width": 35,
      },
      undefined,
    ]
  }
>
  <svg
    data-file-name="SvgIconCart"
    height={30}
    width={30}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F30000",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "height": 20,
          "justifyContent": "center",
          "minWidth": 30,
          "paddingTop": 2,
          "position": "absolute",
          "right": -10,
          "top": -12,
        },
        undefined,
      ]
    }
  >
    <Text
      palette="white"
      size="small"
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "SofiaPro-SemiBold",
            "fontSize": 17,
            "lineHeight": 17,
          },
          undefined,
        ]
      }
      weight="semibold"
    >
      10
    </Text>
  </View>
</View>
`;
